src/main/host/descriptor/socket/inet/tcp.rs

```diff
@@ -614,11 +614,83 @@ impl LegacyTcpSocket {
         errcode.map_err(Into::into)
     }
 
-    pub fn accept(
-        &mut self,
-        _cb_queue: &mut CallbackQueue,
-    ) -> Result<Arc<AtomicRefCell<Self>>, SyscallError> {
-        todo!()
+    pub fn accept(&mut self, _cb_queue: &mut CallbackQueue) -> Result<OpenFile, SyscallError> {
+        let is_valid_listener = unsafe { c::tcp_isValidListener(self.as_legacy_tcp()) } == 1;
+
+        // we must be listening in order to accept
+        if !is_valid_listener {
+            log::debug!("Socket is not listening");
+            return Err(Errno::EINVAL.into());
+        }
+
+        let mut peer_addr: libc::sockaddr_in = unsafe { std::mem::zeroed() };
+        peer_addr.sin_family = libc::AF_INET as u16;
+        let mut accepted_fd = -1;
+
+        // now we can check if we have anything to accept
+        let errcode = Worker::with_active_host(|host| unsafe {
```

Comment:
It's probably reasonable to pass the host and process in to this function explicitly? (and maybe just the whole syscall context object?)

Reply:
I want to avoid this for now since ideally the socket objects shouldn't know anything about the host or process, and future rust versions shouldn't need to. The host and process are only needed for the C TCP socket, so for now I'd rather have them use the globals rather than pass the host/process in as arguments for all the socket types.